server/src/main/java/io/crate/analyze/TableElementsAnalyzer.java

```diff
@@ -224,6 +233,7 @@ public Void visitObjectColumnType(ObjectColumnType<?> node, ColumnDefinitionCont
                     context.logWarnings
                 );
                 columnDefinition.accept(this, childContext);
+                context.currentColumnPosition = childContext.currentColumnPosition;
```

Comment:
Shouldn't the position calculation disappear completely from the coordinator node to fix the concurrent mapping update issue?

Reply:
This is also used by CREATE TABLE so I added a boolean flag and some comments to separate from ADD COLUMN more explicitly.